### PR TITLE
Update revealjs v5 and dependencies 

### LINKED ISSUE #3997: Slide Number Issues with RevealJS's print-pdf and h1 sections
### Bug description

The slide numbering gets messed up by h1 (i.e., `#`) sections when using `print-pdf` with RevealJS. If you create a MWE presentation using the code below, it will correctly number the slides 1 to 4 in the IDE and web browser. But when you go to `http://localhost:6600/?print-pdf` (or whatever your port is) or open the presentation in Chrome and hit `e` on the keyboard, each h1 section will increment the slide number by 2 instead of 1. It will also change back to the `c` format, even if you change it in YAML, and it will add an empty page at the end.

```` qmd
---
format: 
  revealjs:
    slide-number: c/t
    show-slide-number: all
---

# Section 1

## Slide

# Section 2

## Slide

````

Correct display in RStudio IDE: (i.e., `1 / 4`, `2 / 4`, `3 / 4`, `4 / 4`)

image

Incorrect numbering and format in PDF Output: (i.e., `2`, `3`, `5`, `6`, Blank)

image

See the PDF here generated by Chrome
[mwe.pdf](https://github.com/quarto-dev/quarto-cli/files/10439373/mwe.pdf)

---

### Versions

RStudio 2022.12.0+353 "Elsbeth Geranium" Release (7d165dcfc1b6d300eb247738db2c7076234f6ef0, 2022-12-03) for Windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2022.12.0+353 Chrome/102.0.5005.167 Electron/19.1.3 Safari/537.36

Microsoft Windows version 22H2 (OS Build 22623.1095) (64-bit)

Google Chrome version 109.0.5414.75 (Official Build) (64-bit)

Quarto CLI 1.2.269

---

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

### LINKED ISSUE #9699: revealjs export pdf changes code block font
### Bug description

Fonts in code blocks change and line highlighting is deleted

### Steps to reproduce

Visit https://quarto.org/docs/presentations/revealjs/demo/#/line-highlighting look at the font, press e key and scroll down again to see the changed font

### Expected behavior

The font should stay the same

### Actual behavior

The font changes

### Your environment

_No response_

### Quarto check output

_No response_

### PR DESCRIPTION
**Pull request by cderv**
Towards 
- #4795 

- Revealjs 5.1.0, new JS and themes
- Reveal Plugins 4.2.5 for chalkboard 2.3.3

This PR includes docs and process to update revealjs theming. It helps understand how this currently works to maps Sass variables from user's theming, to quarto internals, and then to expected Reveal.js own SASS theming system. 

We probably need more tests as most of the change will have effect in browser running JS code for reveal, so we don't have much test with this yet. 

## Manual testing

- [x] Demo slides: https://quarto.org/docs/presentations/revealjs/demo/ - Works OK ! 
- [x] All simpler demo examples from: https://quarto.org/docs/presentations/revealjs/demo/mini - Works OK ! 
- [x] All examples from https://quarto.org/docs/presentations/revealjs/examples - Works OK ! 
- [x] Checking all the themes change - I made a website to help with this : https://cderv.github.io/quarto-revealjs-themes/

### CODE REVIEW

**Comment by cscheid** [comment]
This looks great!

Keeping in mind the upcoming posit::conf presentations, we should either:

- merge ASAP (as in today)
- or the day after posit::conf is done

What do you think?

**Comment by cderv** [comment]
😄 definitely the main question. I ask myself the same, hence the draft state. We have a stable release working we could also consider that anyone doing a presentation for "production" (posit::conf) would use a stable version and not a pre-release 😉 Though our users likes to live on the edge of new stuff ! 😆 

I don't think the effect if big, but we don't really have test regarding how feature works in in a live browser situation so it requires manual testing. And I am just starting them. 

Could be small or no effect, but could also be the opposite 🤷‍♂️. So considering the milestone of conf, we should probably let this in a branch. This also leave me time to add to this, rebasing often on main and adding some tests. 

What is your take on this ?

**Comment by cscheid** [comment]
You have the hands-on experience.

> So considering the milestone of conf, we should probably let this in a branch.

That works for me. Let's plan to merge right after conf.

**Comment by cderv** [comment]
First issue to solve 
````markdown
---
title: "title"
format: 
  revealjs:
    chalkboard: true
---

## Quarto

Quarto enables you to weave together content and executable code into a finished document. To learn more about Quarto see <https://quarto.org>.

````

image

We get blurry screen by default following the update of chalkboard. 😭

**Comment by cderv** [comment]
Nevermind, it seems it is fixed upstream: 
- https://github.com/rajgoel/reveal.js-plugins/pull/180

We need an unreleased version of chalkboard.

**Comment by cderv** [comment]
Very small theme change after all, and it is now more inline with revealjs own theme. 

All good to merge this PR so I'll go ahead. 

I'll update documentation for 1.6 to close #4795 , including adding an highlight

File: tests/docs/smoke-all/2024/05/03/9548.qmd
[the remark this answers is not in the document]
Comment (cscheid) [reply tests/docs/smoke-all/2024/05/03/9548.qmd:1]: Ok. That's unfortunate, but currently worth the additional pain of checking the test.

Let's keep an eye on this kind of thing, just so we don't let ourselves get bogged down with lots of manual updates on tests.

**Pull request merged**